alodataset/kitti_odometry.py

```diff
@@ -0,0 +1,288 @@
+import os
+import torch
+import numpy as np
+from typing import List, Dict, Any
+
+from alodataset import BaseDataset, SplitMixin, Split
+from aloscene import Frame, Pose, CameraIntrinsic, CameraExtrinsic
+
+
+def sequence_index(start, seq_size, skip):
+    end = start + (seq_size - 1) * (skip + 1)
+    return np.linspace(start, end, seq_size).astype(int).tolist()
+
+
+def sequence_indices(n_samples, seq_size, skip, seq_skip):
+    for start in range(0, n_samples - (seq_size - 1) * (skip + 1), seq_skip + 1):
+        yield sequence_index(start, seq_size, skip)
+
+
+class KittiOdometryDataset(BaseDataset, SplitMixin):
+    def __init__(
+        self,
+        name="kitti_odometry",
+        sequences=None,
+        grayscale=True,
+        right_frame=True,
+        sequence_size=3,
+        skip=1,
+        sequence_skip=1,
+        **kwargs,
+    ):
+        super().__init__(name=name, **kwargs)
+
+        self.grayscale = grayscale
+        self.right_frame = right_frame
+        self.sequence_size = sequence_size
+        self.skip = skip
+        self.sequence_skip = sequence_skip
+
+        if self.sample:
+            return
+
+        self.dataset_dir = os.path.join(self.dataset_dir, "dataset")
+
+        self.sequences = self._load_sequences(sequences)
+
+        self.items: Dict[Any, Any] = {}
+
+        self.seq_params = {}
+
+        for seq in self.sequences:
+
+            # Exploit data from calib.txt
+            calib = self._load_calib(os.path.join(self.dataset_dir, "sequences", seq, "calib.txt"))
+
+            # Register sequence parameters
+            self.seq_params["baseline"] = calib["b_gray" if self.grayscale else "b_rgb"]
+            self.seq_params["left_intrinsic"] = CameraIntrinsic(
+                np.c_[calib[f"K_cam{0 if grayscale else 2}"], np.zeros(3)]
+            )
+            self.seq_params["left_extrinsic"] = CameraExtrinsic(calib[f"T_cam{0 if grayscale else 2}_rect"])
+            if right_frame:
+                self.seq_params["right_intrinsic"] = CameraIntrinsic(
+                    np.c_[calib[f"K_cam{1 if grayscale else 3}"], np.zeros(3)]
+                )
+                self.seq_params["right_extrinsic"] = CameraExtrinsic(calib[f"T_cam{1 if grayscale else 3}_rect"])
```

Comment:
A different calibration file is read for each sequence. 
Therefore, `seq_params` should be indexed by the sequence name.

For example :
```
# loading the left intrinsic matrix for the sequence `seq`
cam_intrinsic = self.seq_params[seq]["left_intrinsic"]
```